tests/boot/with_mcumgr/sysbuild.cmake

```diff
@@ -0,0 +1,16 @@
+# Add the mcuboot key file to the secondary swapped app
+# This must be done here to ensure that the same key file is used for signing
+# both the primary and secondary apps
+set(smp_svr_CONFIG_MCUBOOT_SIGNATURE_KEY_FILE
+    \"${SB_CONFIG_BOOT_SIGNATURE_KEY_FILE}\" CACHE STRING
+    "Signature key file for signing" FORCE)
+
+# Add the swapped app to the build
+ExternalZephyrProject_Add(
+  APPLICATION smp_svr
+  SOURCE_DIR ${ZEPHYR_BASE}/samples/subsys/mgmt/mcumgr/smp_svr
+)
+
+# Do not add the swapped app to the list of images to flash,
```

Comment:
Now I see why it is not getting the signing key. This is not acceptable, it's hacking around sysbuild and calling things that sysbuild itself should be calling, understandable that you don't want to flash the image but need another way of doing it which sysbuild natively supports, not doing it by calling internal sysbuild cmake functions (I will take the task of looking into this)

Reply:
I still don't understand. Let me explain:
- APP + MCUboot are built, signed and used to flash (like in [samples/application_development/sysbuild/with_mcuboot](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/application_development/sysbuild/with_mcuboot)
- `smp_svr` app is added as a third app, it is also built and signed, but **not added to the list of images to flash**. I used sysbuild here, just to have ready artefacts to use during test procedure: I need only `zephyr.signed.bin`, to send it with `mcumgr` and use to upgrade.
- `smp_svr` is added in similar way like the third app in  [tests/boot/test_mcuboot](https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/boot/test_mcuboot), but there it is also added to the list of images to flash and is used when the image swap is requested. But this test is now skipped, because of #58080 

If using sysbuild to build a new app is a "hack" and is not acceptable, then I can remove third app (smp_svr) from sysbuild, then (in pytest scenario, after flashing) prepare image for upgrade using `west sign` (take `zephyr.signed.bin` from original APP and sign it with another version). But I think this will be also a bit tricky.

Reply:
The problem is by doing this, you are skipping lines 66 - 96 of https://github.com/zephyrproject-rtos/zephyr/blob/main/share/sysbuild/CMakeLists.txt#L66 when really, only line 95 should be skipped

Reply:
@nordicjm IMO the question is what process we want to test and what must be done with sysbuild (maybe it was already answered?). Obviously, sysbuild must be used for building the bootloader and the application image. But in the process the same app image is used, the only difference is in the signing. Hence, is it mandatory to do duplicated builds with sysbuild just to have different "sign"?

Reply:
@nordicjm I removed the `smp_svr`, now only APP + MCUboot are built with sysbuild. The image for upgrade is prepared now using `west sign` command, during pytest scenario.